include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -54,7 +54,7 @@ __pattern_walk1(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _ForwardIt
 
     oneapi::dpl::__par_backend_hetero::__parallel_for(
         _BackendTag{}, __exec, unseq_backend::walk_n<_ExecutionPolicy, _Function>{__f}, __n, __buf.all_view())
-        .wait();
+        .__deferrable_wait();
```

Comment:
I cant currently find any problems with this one currently in the repo, but it is to some extent a "building block", which other algorithms may use internally as a piece of a multi-kernel operation which may have internal dependencies.  

This seems like a possible source of errors future the way it is currently implemented.
It may be worth it to do something like what is done in `__patern_walk2` to provide control of `_IsSync` to the caller, then use `ONEDPL_ALLOW_DEFERRED_WAITING` to control that template variable in the cases which make sense.  Alternatively, we could allow algorithms to explicitly set this pattern to synchronous, and otherwise, by default depend on `__deferrable_wait()`.

Reply:
We discussed and think its best to leave this for now prior to release, and perhaps make more changes after release to switch some patterns to a system like __pattern_walk2.